### PR TITLE
Configuration for AWS codebuild windows build

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: buildspec-windows.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        Remove-Item cbmc-cpp -Force -Recurse
+        Remove-Item goto-gcc -Force -Recurse
+        Remove-Item systemc -Force -Recurse
+        Remove-Item goto-instrument\slice08 -Force -Recurse
+        Remove-Item goto-analyzer/constant_propagation_nondet_rounding_mode -Force -Recurse
+        cd ..
+
+      - |
+        $env:Path = "C:\tools\cygwin\bin;$env:Path"
+        cmd /c 'call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x64 && bash -c "make -C regression test" '
```
Comment (tautschnig) [inline buildspec-windows.yml:63]: Shouldn't the JBMC regression tests also be run?

File: buildspec-windows.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+version: 0.2
+
+phases:
+  install:
+    commands:
+      - choco install cyg-get -y --no-progress
+      - cyg-get bash patch bison flex make wget perl
```
Comment (tautschnig) [inline buildspec-windows.yml:7]: Would `curl` also be available? Just curious as I'm wondering whether we can consistently move towards `curl` for all download steps across all platforms.

File: src/goto-instrument/cover_instrument_mcdc.cpp [hunk 3]
```diff
@@ -14,6 +14,7 @@ Author: Daniel Kroening
 #include <langapi/language_util.h>
 
 #include <algorithm>
+#include <iterator>
```
Comment (tautschnig) [inline src/goto-instrument/cover_instrument_mcdc.cpp]: Procedural note: if this commit is moved to a separate PR, the CODEOWNERS-induced reviewer set will be a bit more reasonable.

**Review by tautschnig — APPROVED** [review]

File: src/goto-instrument/cover_instrument_mcdc.cpp (hunk 3 above)
Comment (kroening) [reply src/goto-instrument/cover_instrument_mcdc.cpp]: Yes, but PR dependencies don't really exist...

File: buildspec-windows.yml (hunk 2 above)
Comment (kroening) [reply buildspec-windows.yml:7]: Possibly, trying.

File: buildspec-windows.yml (hunk 1 above)
Comment (kroening) [reply buildspec-windows.yml:63]: There's some problem with jar packing, still tweaking.

File: src/goto-instrument/cover_instrument_mcdc.cpp (hunk 3 above)
Comment (tautschnig) [reply src/goto-instrument/cover_instrument_mcdc.cpp]: I'm aware, but it could likely get reviewed and merged within minutes so that is in well before the build configuration.

File: buildspec-windows.yml (hunk 1 above)
Comment (peterschrammel) [reply buildspec-windows.yml:63]: Unit tests should be run too, both from CBMC and JBMC

**Comment by kroening** [comment]
curl appears to work; if this is superior to wget or lwp-download, we could go for it.
https://daniel.haxx.se/docs/curl-vs-wget.html

File: src/goto-instrument/cover_instrument_mcdc.cpp (hunk 3 above)
Comment (kroening) [reply src/goto-instrument/cover_instrument_mcdc.cpp]: #2312

**Review by peterschrammel — APPROVED** [review]
This can be rebased now. 

I insist that the unit tests need to be run as well (both on Windows and Linux configurations). To get this merged now, I've added this issue https://github.com/diffblue/cbmc/issues/2328
